core/common/src/main/scala/org/typelevel/otel4s/context/Key.scala

```diff
@@ -45,4 +53,17 @@ object Key {
         p: Provider[F, K]
     ): Provider[F, K] = p
   }
+
+  @inline private def defaultShowKey(key: Key[_]): String =
```

Comment:
The `@inline` annotation is effective  when the `-opt:l:inline` compiler flag is enabled. 

If I recall it right, we don't enable it in libraries, because the generated bytecode may differ.

Reply:
we don't even for code within our own library? interesting. I know the stdlib does

Reply:
@armanbilge could you shed some light here? I don't remember the exact reason, unfortunately

Reply:
I checked this article https://docs.scala-lang.org/overviews/compiler-options/optimizer.html regarding the inlining optimization.

The `@inline` annotation is used in several places across the Typelevel ecosystem, primarily in the Scala Native and ScalaJS code. For example, cats-effect [ByteStack](https://github.com/typelevel/cats-effect/blob/04cc47ba7a61f5352a3eafecba79125ed33f9e3f/core/js/src/main/scala/cats/effect/ByteStack.scala#L25).

And Arman said the following:
> we'd have to enable compiler flags for `@inline` to do something on JVM anyway, currently we don't use them and just rely on JIT
> the places where you see `@inline` are JS or Native specific code, because their linkers respect it 


I come up with the following conclusions:
- Let's keep the `@inline` annotation
- Scala Native and ScalaJS linkers will inline the call
- JIT compiler will inline the calls on JVM
- An end user of the library (a.k.a. application) can enable global inlining `-opt:l:inline:**`

Reply:
What do you think @NthPortal?